paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheet.kt

```diff
@@ -1516,6 +1535,67 @@ class PaymentSheet internal constructor(
         }
     }
 
+    /**
+     * Options to block certain card brands on the client
+     */
+    sealed class CardBrandAcceptance : Parcelable {
+
+        /**
+         * Card brand categories that can be allowed or disallowed
+         */
+        @Parcelize
+        @ExperimentalCardBrandFilteringApi
+        enum class BrandCategory : Parcelable {
+            /**
+             * Visa branded cards
+             */
+            Visa,
+
+            /**
+             * Mastercard branded cards
+             */
+            Mastercard,
+
+            /**
+             * Amex branded cards
+             */
+            Amex,
+
+            /**
+             * Discover branded cards
+             * **Note**: Encompasses all of Discover Global Network (Discover, Diners, JCB, UnionPay, Elo).
+             */
+            Discover
+        }
+
+        /**
+         * Accept all card brands supported by Stripe
+         */
+        @Parcelize
+        @ExperimentalCardBrandFilteringApi
+        data object All : CardBrandAcceptance()
+
+        /**
+         * Accept only the card brands specified in the associated value
+         * **Note**: Any card brands that do not map to a `BrandCategory` will be blocked when using an allow list.
+         */
+        @Parcelize
+        @ExperimentalCardBrandFilteringApi
+        data class Allowed(
+            val brands: List<BrandCategory>
```

Comment:
I think this val and the one in Disallowed can be internal.

That will allow merchants to create the object, but not reference the list they passed in.

Reply:
Ah good idea!